numpy/core/_add_newdocs_scalars.py

```diff
@@ -283,6 +283,9 @@ def add_newdoc_for_scalar_type(obj, fixed_aliases, doc):
     ``1970-01-01T00:00:00``.
     If created from string, the string can be in ISO 8601 date
     or datetime format.
+    
+    When parsing a string to create a datetime object, if the string contains
+    a timezone, the timezone will be dropped.
```

Comment:
What's the exact logic?  A userwarning is given (I think it would be good to mention),  we do have a Warnings heading in principle, too.

Can we explain (and check) what we actually do?  And do we actually want to keep it for all of them?
* Timezones differences without the Z are simply applied
* A Z (and I guess also trailing) is simply ignored, the naive timezone is assumed to be UTC?

Is that correct?